src/usecases/connect-figma-team-use-case.ts

```diff
@@ -52,7 +59,11 @@ export const connectFigmaTeamUseCase = {
 			return figmaTeam.toFigmaTeamSummary();
 		} catch (e) {
 			if (e instanceof UnauthorizedFigmaServiceError) {
-				throw new ForbiddenByFigmaUseCaseResultError({ cause: e });
+				if (e.cause instanceof BadRequestHttpClientError) {
```

Comment:
**suggestion:** It could be benefitial to handle `BadRequestHttpClientError` in `FigmaService.createFileUpdateWebhook`, so:

1.  `FigmaService.createFileUpdateWebhook` handles `BadRequestHttpClientError` and throw a more specific `PaidPlanRequiredFigmaServiceError`.
2. `connectFigmaTeamUseCase` handles the very specific `PaidPlanRequiredFigmaServiceError`.

This can bring the following benefits:

- `connectFigmaTeamUseCase` does not know about what errors my _transient_ dependencies (dependencies of my dependencies) throw. Therefore, it handles only errors that are part of contract of its _direct_ dependencies. We try to document errors that are part of the function/method contract: e.g., https://github.com/atlassian-labs/figma-for-jira/blob/3100c153d54723111745fd5829d41d42b35fda30/src/infrastructure/figma/figma-service.ts#L224.
- It allows to handle `HTTP 400` specifically from `POST /v2/webhooks`. Currently, the code catch all `HTTP 400` regardless whether they are thrown from Figma API or Jira API.

I will try to address this to get the PR merged ASAP.